lib/routes/emails.js

```diff
@@ -80,17 +80,28 @@ module.exports = (log, db, mailer, config, customs, push) => {
             name: 'recovery_email_reason.push'
           })
         }
-
         cleanUpIfAccountInvalid()
           .then(createResponse)
           .then(reply, reply)
 
         function cleanUpIfAccountInvalid () {
-          // Some historical bugs mean we've allowed creation
-          // of accounts with invalid email addresses. These
-          // can never be verified, so the best we can do is
-          // to delete them so the browser will stop polling.
           if (! sessionToken.emailVerified) {
+            const now = new Date().getTime()
+            const staleTime = now - config.emailStatusPollingTimeout
+
+            if (sessionToken.createdAt < staleTime) {
+              log.info({
+                op: 'recovery_email.status',
```

Comment:
todo: let's log both emailVerified and tokenVerified, we need to take this out of the ` if (! sessionToken.emailVerified) {` block. Otherwise we would miss verified emails that do no have a verified session.